src/variant.ts

```diff
@@ -234,18 +231,14 @@ export class VariantManager implements vscode.Disposable {
   private async _reloadVariantsFile(filepath?: string) {
     const validate = await loadSchema('schemas/variants-schema.json');
 
-    const workdir = util.getPrimaryWorkspaceFolder();
-    if (!workdir) {
-      // Can't read, we don't have a dir open
-      return;
-    }
+    const workdir = this.folder.uri.fsPath;
```

Comment:
Is `this.folder` always an object?

Reply:
Yes. There's one variant instance for each folder, and folder is passed in the constructor. It is only constructed in cmake-tools.ts line 153. Since there must be a folder for CMakeTools, this.folder is always an object.

Reply:
What I meant was whether this can ever be `null` or `undefined`.  (i.e. if the user activates CMake Tools without opening a folder)

Reply:
Unlike before, each CMakeTools is bonded to a folder now. So it won't be null or undefined.

Reply:
Thanks.  That's what I wanted to know.